docs/service-fabric/refactor-migrated-app.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Refactoring an Azure Service Fabric Application migrated from Azure Cloud Services 
+description: How to refactor an existing Azure Service Fabric Application migrated from Azure Cloud Services
+author: petertay
+ms.date: 01/30/2018
+---
+# Refactoring an Azure Service Fabric Application migrated from Azure Cloud Services
+
+[![GitHub](../_images/github.png) Sample code][sample-code]
+
+This article describes refactoring an existing Azure Service Fabric application to a more granular architecture. This article focuses on the design, packaging, performance, and deployment considerations of the refactored Service Fabric application.
+
+## Scenario
+
+As discussed in the previous article, [migrating an Azure Cloud Services application to Azure Service Fabric][migrate-from-cloud-services], in 2012 the patterns & practices team documented the process for designing and implementing a Cloud Services application in Azure. The book describes a fictitious company named Tailspin that wants to create a Cloud Services application named **Surveys**. The Surveys application allows users to create and publish surveys that can be answered by the public. The following diagram shows the architecture of this version of the Surveys application:
+
+![](./images/surveys_01.png)
+
+The **Tailspin.Web** web role hosts an ASP.NET MVC site that Tailspin customers use to:
+* sign up for the Surveys application,
+* create or delete a survey,
+* view results for a survey,
+* request that survey results be exported to SQL, and
+* view aggregated survey results and analysis.
+
+The **Tailspin.Web.Survey.Public** web role also hosts an ASP.NET MVC site that the public visits to fill out the surveys. Their responses are put in a queue to be saved.
+
+The **Tailspin.Workers.Survey** worker role performs background processing by picking up requests from multiple queues.
+
+Last year, the patterns & practices team created a new project to port this application to Azure Service Fabric. The goal of this project was to make only the necessary code changes to get the application running in an Azure Service Fabric cluster. As a result, the original web and worker roles were not decomposed into a more granular architecture. The resulting architecture is very similar to the Cloud Service version of the application:
+
+![](./images/surveys_02.png)
+
+The **Tailspin.Web** service is ported from the original *Tailspin.Web* web role.
+
+The **Tailspin.Web.Survey.Public** service is ported from the original *Tailspin.Web.Survey.Public* web role.
+
+The **Tailspin.AnswerAnalysisService** service is ported from the original *Tailspin.Workers.Survey* worker role.
+
+> [!NOTE] 
+> While minimal code changes were made to each of the web and worker roles, **Tailspin.Web** and **Tailspin.Web.Survey.Public** were modified to self-host a Kestrel web server. 
```

Comment:
Link to Kestrel? Or say something like "self host a web server without IIS"? Readers may not be familiar with Kestrel.